### ISSUE TITLE
Question - can we do one retry using `retries`

### ISSUE THREAD

**Issue by alexzaytsev-newsroomly** [immortal/immortal#42]
We use immortal to configure some code insider our docker containers.
Depending on some external configuration, we may want our service to always be restarted, or to run just once. 

I tried setting `retries` to 0 in immortal configuration but that didn't seem to do it.

Any way i can use immortal so that supervisor exits if service exits, using `run.yml` approach?

**Comment by alexzaytsev-newsroomly** [comment]
Actually updating to immortal 0.20 and setting `IMMORTAL_EXIT` variable almost did it - except on exit immortal crashes with the following go stack trace:

```
2018/09/11 06:08:31 PID 25 (/some/executable) terminated, exit status 0 [1.484205s user 99.079ms sys 1m4.953751881s up]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4ff1ce]
goroutine 1 [running]:
log/syslog.(*Writer).Close(0x0, 0x0, 0x0)
/usr/local/go/src/log/syslog/syslog.go:180 +0x2e
main.main()
/go/src/github.com/immortal/immortal/cmd/immortal/main.go:125 +0x8f8
```

**Comment by nbari** [comment]
Hi @alexzaytsev-newsroomly, `retries 1`  should do the work, starts the process and if it fails it will restart it only once , `retries 0`  is like running the app without supervisor but to be concistent could be probably nice to have since it will also be the only way to exit if the apps exists.

Could you describe more how you are using the `IMMORTAL_EXIT` ,  I would like to test more in detail about why is crashing. thanks in advance 👍

**Comment by nbari** [comment]
Problem seems to be here: https://github.com/immortal/immortal/blob/master/supervise.go#L94-L103

**Comment by nbari** [comment]
@alexzaytsev-newsroomly can you reproduce the "crash"? I am currently testing with `retries 0`, `-r 0` and `IMMORTAL_EXIT` but working as expected

**Comment by nbari** [comment]
@alexzaytsev-newsroomly  please give a try to the develop branch

**Comment by nbari** [comment]
Closing this for now, please feel free to reopen the issue if persists
